src/hardware/joystick.cpp

```diff
@@ -404,6 +413,131 @@ void JOYSTICK_ParseConfiguredType()
+	}
+}
+
+static void calibrate_y_scalar_down(const bool pressed)
+{
+	calibrate_axis(pressed, calibrated_axis_rates.y, NegativeScale);
+}
+
+static void calibrate_y_scalar_up(const bool pressed)
+{
+	calibrate_axis(pressed, calibrated_axis_rates.y, PositiveScale);
+}
+
+static void calibrate_y_offset_down(const bool pressed)
+{
+	calibrate_axis(pressed, calibrated_axis_rates.y, NegativeOffset);
+}
+static void calibrate_y_offset_up(const bool pressed)
+{
+	calibrate_axis(pressed, calibrated_axis_rates.y, PositiveOffset);
+}
+static void reset_y_calibration(const bool pressed)
+{
+	if (pressed) {
+		calibrated_axis_rates.y = default_axis_rates.y;
+		LOG_MSG("JOYSTICK: y_calibration = auto");
+	}
+}
+
+static void activate_calibration_hotkeys()
+{
+	LOG_MSG("JOYSTICK: Activating joystick calibration hotkeys");
+	// clang-format off
+	MAPPER_AddHandler(&calibrate_x_scalar_left,  SDL_SCANCODE_LEFT,   PRIMARY_MOD, "jxsl", "JXSL");
+	MAPPER_AddHandler(&calibrate_x_scalar_right, SDL_SCANCODE_RIGHT,  PRIMARY_MOD, "jxsr", "JXSR");
+	MAPPER_AddHandler(&calibrate_x_offset_left,  SDL_SCANCODE_LEFT,   MMOD2,       "jxol", "JXOL");
+	MAPPER_AddHandler(&calibrate_x_offset_right, SDL_SCANCODE_RIGHT,  MMOD2,       "jxor", "JXOR");
+	MAPPER_AddHandler(&reset_x_calibration,      SDL_SCANCODE_DELETE, PRIMARY_MOD, "jxrs", "JXRS");
+	MAPPER_AddHandler(&calibrate_y_scalar_down,  SDL_SCANCODE_DOWN,   PRIMARY_MOD, "jysd", "JYSD");
+	MAPPER_AddHandler(&calibrate_y_scalar_up,    SDL_SCANCODE_UP,     PRIMARY_MOD, "jysu", "JYSU");
+	MAPPER_AddHandler(&calibrate_y_offset_down,  SDL_SCANCODE_DOWN,   MMOD2,       "jyod", "JYOD");
+	MAPPER_AddHandler(&calibrate_y_offset_up,    SDL_SCANCODE_UP,     MMOD2,       "jyou", "JYOU");
+	MAPPER_AddHandler(&reset_y_calibration,      SDL_SCANCODE_HOME,   PRIMARY_MOD, "jyrs", "JYRS");
+	// clang-format off
+}
+
+static void configure_calibration(const Section_prop &settings)
+{
+	if (settings.Get_bool("use_calibration_hotkeys"))
+		activate_calibration_hotkeys();
+
+	auto axis_rates_from_pref = [](const std::string_view pref,
+	                               const AxisRateConstants &default_rates) {
+		if (AxisRateConstants parsed_rates = {};
+		    sscanf(pref.data(), "%lf,%lf", &parsed_rates.scalar, &parsed_rates.offset) == 2) {
+			LOG_MSG("JOYSTICK: Loaded custom %c-axis calibration parameters (%.6g,%.6g)",
+			        default_rates.axis,
+			        parsed_rates.scalar,
+			        parsed_rates.offset);
+			return parsed_rates;
+		}
+		if (pref != "auto" && pref.length() != 0)
+			LOG_WARNING("JOYSTICK: Invalid %c_calibration parameters: %s."
+			            " Must be auto or number,number.",
+			            default_rates.axis,
+			            pref.data());
+		return default_rates;
+	};
+	const auto x_cal_pref   = settings.Get_string("x_calibration");
+	calibrated_axis_rates.x = axis_rates_from_pref(x_cal_pref, default_axis_rates.x);
+
+	const auto y_cal_pref   = settings.Get_string("y_calibration");
```

Comment:
I'd prefer `joy_x_calibration` and `joy_y_calibration`, given we don't have proper scoping in the config.

Reply:
indeed :+1: 
fixed.